src/Microsoft.TestPlatform.Common/DataCollection/DataCollectorLoader.cs

```diff
@@ -29,18 +57,102 @@ public DataCollector Load(string location, string assemblyQualifiedName)
 #else
                 assembly = AssemblyLoadContext.Default.LoadFromAssemblyPath(location);
 #endif
-                var dataCollectorType = assembly.GetTypes().Where(t => t.AssemblyQualifiedName.Equals(assemblyQualifiedName)).FirstOrDefault();
-                dataCollectorInstance = Activator.CreateInstance(dataCollectorType) as DataCollector;
+
+                foreach (var type in assembly.GetTypes())
```

Comment:
This is not following the convention used by other extension loaders. Is that intentional?